File: crawler/src/main/java/com/project/crawler/entity/TechBlog.java
package com.project.crawler.entity;

import com.project.crawler.constants.Source;
import com.project.crawler.constants.Status;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.LocalDateTime;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@Table(name = "tech_blog")
@NoArgsConstructor
public class TechBlog {

    @Id
    private Long id;

    private String url;

    @Enumerated(EnumType.STRING)
    private Status status;

    @Enumerated(EnumType.STRING)
    private Source source;

    private LocalDateTime publishedAt;

    public void markAsUnpublished() {
        this.status=Status.UNPUBLISHED;
    }
    public void markAsFailed() {
        this.status=Status.FAILED;
    }

    @Builder
    public TechBlog(Long id, String url, Status status,LocalDateTime publishedAt) {
        this.id=id;
        this.url=url;
        this.status=status;
        this.publishedAt=publishedAt;
    }
}

File: batch/src/main/kotlin/com/project/batch/domain/TechBlog.kt
package com.project.batch.domain

import com.project.batch.constants.Region
import com.project.batch.constants.Status
import lombok.Getter
import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table
import java.time.Instant

@Table("tech_blog")
@Getter
data class TechBlog(
    @Id val id: Long,
    val source: String,
    val region: Region,
    val title: String,
    val url: String,
    val publishedAt: Instant,
    val tags: List<String>,
    val status: Status = Status.PENDING,
)